docs/modules/ROOT/pages/patterns.adoc

```diff
@@ -0,0 +1,216 @@
+```
+
+```cypher
+MATCH (this0:Person)-[this1:ACTED_IN]->(this2:Movie)
+```
+
+=== Relationship direction
+
+By default, a relationship in a pattern will be created as a left-to-right pattern. The direction can be changed when defining the relationship in the pattern using the method `withDirection`:
+
+```javascript
+const pattern = new Cypher.Pattern(person).related(actedIn).withDirection("left").to(movie);
+```
+
+This translates to the following Pattern:
+
+```cypher
+(this0:Person)<-[this1:ACTED_IN]-(this2:Movie)
+```
+
+The possible options for `withDirection` are:
+
+* `right` (default): A left-to-right (`()-[]->()`) pattern.
+* `left`: A right-to-left (`()<-[]-()`) pattern.
+* `undirected`: An undirected (`()-[]-()`) pattern.
+
+=== Remove variable names
+
+By default, a pattern will render all variable names of the Node and Relationship variables, but it is possible to hide them from particular elements of the pattern with the method `withoutVariable`:
+
+```javascript
+const pattern = new Cypher.Pattern(person).related(actedIn).withoutVariable().to(movie).withoutVariable();
+```
+
+This translates to:
+
+```cypher
+(this0:Person)-[:ACTED_IN]->(Movie)
+```
+
+=== Remove labels and types
+
+Labels can be hidden from nodes with the method `withoutLabels`:
+
+
+```javascript
+const pattern = new Cypher.Pattern(person).withoutLabels().related(actedIn).to(movie);
+```
+
+```cypher
+(this0)-[this1:ACTED_IN]->(this1:Movie)
+```
+
+For relationships, however, you must use `withoutType`:
+
+```javascript
+const pattern = new Cypher.Pattern(person).related(actedIn).withoutType().to(movie);
+```
+
+This translates to:
+
+```cypher
+(this0:Person)-[this1]->(this1:Movie)
+```
+
+[NOTE]
+====
+Alternatively, labels and types are optional when defining a Node or a Relationship. Unless you need to reuse the same variable, you can skip defining the labels altogether.
+====
+
+=== Properties
+
+Patterns may contain properties to match, both on the node and relationships. These can be added using the `withProperties` method:
+
+```javascript
+const pattern = new Cypher.Pattern(person)
+            .withProperties({
+                name: new Cypher.Param("Person")
+            })
+            .related(actedIn)
+            .to(movie);
+```
+
+```cypher
+(this0:Person { name: $param0 })-[this1:ACTED_IN]->(this2:Movie)
+```
+
+`withProperties` takes an object with the properties to match and the Param objects to be used as the expected values of the pattern. `withProperties` can be used in both nodes and relationship elements.
+
+=== Longer patterns
+
+Patterns are not limited to a node or a relationship. Patterns can be arbitrarily long. For example:
+
+```javascript
+const user = new Cypher.Node({labels: ["User"]});
+
+const pattern = new Cypher.Pattern(person).related(actedIn).to(movie).related().withDirection("left").to(user);
+```
+
+```cypher
+(this0:Person)-[this1:ACTED_IN]->(this2:Movie)<-[this3]-(this4:User)
+```
+
+== Advanced patterns
+
```

Comment:
It would be good to have an introductory paragraph here. Perhaps what you put under longer patterns could be placed here instead?